### ISSUE TITLE
An "expected an expression" error in compiled CUDA code.

### ISSUE THREAD

**Issue by LukaHorvat** [AccelerateHS/accelerate#318]
I have the following Haskell function

```
mod'' :: Exp Double -> Exp Double -> Exp Double
mod'' x y = x - (fromIntegral (round (x / y) :: Exp Int) * y)
```

(never mind what it does).

This gets compiled to something like

```
const double v38 = lv214 + ({ const double v35 = 1.23456789e8 * lv214;  ; v35 - 2.0 * (double) (Int64) round(v35 / 2.0); }) * lv24;
```

which is apparently an error.

**Comment by tmcdonell** [comment]
What is the error message? What version of ghc/gcc/nvcc etc. are you using? Are you on windows/linux/mac? Some information to help diagnose and work out the problem would be useful here...

**Comment by LukaHorvat** [comment]
You're right. Sorry. The error is the one in the title. Just that.

I'm running GHC 7.10.3 on Windows. GCC is the one that comes with that
version of GHC.

I'll check the nvcc version when I get home.

**Comment by LukaHorvat** [comment]
```
nvcc: NVIDIA (R) Cuda compiler driver
Copyright (c) 2005-2015 NVIDIA Corporation
Built on Tue_Aug_11_14:49:10_CDT_2015
Cuda compilation tools, release 7.5, V7.5.17
```

**Comment by tmcdonell** [comment]
I suspected that you were using windows. The problem is that MSVC does not support statement expressions. We discussed this in #234, and @mwu-tow has a fix there that should also work for you.

It might be worthwhile trying to merge that fix into accelerate-cuda (I think the necessary changes to language-c-quote eventually made it in) I will just have to reacquaint myself with everything that was discussed etc. The long term plan is probably to just transition everybody to the LLVM backends, where this problem doesn't exist.
